flink-table-store-core/src/main/java/org/apache/flink/table/store/file/utils/FileStorePathFactory.java

```diff
@@ -106,10 +115,12 @@ public SstPathFactory createSstPathFactory(BinaryRowData partition, int bucket)
     }
 
     public String getPartitionString(BinaryRowData partition) {
-        return PartitionPathUtils.generatePartitionPath(
-                partitionComputer.generatePartValues(
-                        Preconditions.checkNotNull(
-                                partition, "Partition row data is null. This is unexpected.")));
+        synchronized (partitionComputer) {
```

Comment:
Why needs add `synchronized`?

Reply:
Because we have to make sure that this class is thread safe. Partition computer is not thread safe.

Another option is to create a read-only path factory which only supports a limited number of methods.

Reply:
After discussion we decided to mark some methods as thread-safe while others as not-thread-safe.